src/main/resources/com/devonfw/ide/sonarqube/common/rules/devon4j/C2_java.html

```diff
@@ -1,5 +1 @@
-<p>This rule checks that all components only depend on components that were declared as dependencies in the architecture.json.</p>
-<h2>Noncompliant Code Example</h2>
-<pre></pre>
-<h2>Compliant Solution</h2>
-<pre></pre>
\ No newline at end of file
+<p>This rule checks that all components only depend on components that were declared as dependencies in the architecture.json.</p>
```

Comment:
This change does not seem to relate to the task of this PR. Did you intend to remove parts from this documentation?

Reply:
Yes, that was done willingly. It's true that it does not belong in this PR, but I became aware of some issues when I wrote the html file for this rule, so I just made those changes here. It's not a problem if I should revert those changes and raise another PR for them though.

In C2_java.html, I deleted the headers for the code examples because I think code examples wouldn't fit here. What could be possible here is a reference to the wiki of our plugin where we show how to properly declare components in the architecture.json.

In the html files of P1, E1 and E4 I just fixed a typo and added some linebreaks.

Reply:
Fine for the moment. No need to change now and thanks for the rationale.
We can later improve these documentation.